WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/CardPresentPaymentsOnboardingIPPUsersRefresher.swift

```diff
@@ -0,0 +1,45 @@
+import CoreData
+import Yosemite
+import Storage
+
+/// Refreshes the CPP onboarding state if there are IPP transactions stored
+///
+class CardPresentPaymentsOnboardingIPPUsersRefresher {
+    private let storageManager: StorageManagerType
+    private let cardPresentPaymentsOnboardingUseCase: CardPresentPaymentsOnboardingUseCaseProtocol
+
+    private lazy var ordersResultsController: ResultsController<StorageOrder> = {
+        return ResultsController<StorageOrder>(storageManager: storageManager, sortedBy: [])
+    }()
+
+    init(storageManager: StorageManagerType = ServiceLocator.storageManager,
+         cardPresentPaymentsOnboardingUseCase: CardPresentPaymentsOnboardingUseCaseProtocol = CardPresentPaymentsOnboardingUseCase()) {
+        self.storageManager = storageManager
+        self.cardPresentPaymentsOnboardingUseCase = cardPresentPaymentsOnboardingUseCase
+
+        try? ordersResultsController.performFetch()
+    }
+
+    func refreshIPPUsersOnboardingState() {
+        guard usedIPPBefore() else {
+            return
+        }
+
+        cardPresentPaymentsOnboardingUseCase.refresh()
+    }
+}
+
+private extension CardPresentPaymentsOnboardingIPPUsersRefresher {
+    func usedIPPBefore() -> Bool {
+        let IPPTransactionsFound = ordersResultsController.fetchedObjects.filter({
+            $0.customFields.contains(where: { $0.key == Constants.receiptURLKey })})
```

Comment:
This might be silly, but could we base this decision on what's already cached?

e.g. a `case .neverStarted` Onboarding state that prevents refreshing, or only refresh if there's something non-nil in the cache?

It feels like this shouldn't work, so please look for flaws!

Reply:
Changed in https://github.com/woocommerce/woocommerce-ios/pull/9002/commits/f2d935651943c46fef23c6cb9a22acd4aee84efa